### DISCUSSION TITLE
Installing 2.3.40 on sensors hangs for a long time if no proxy is given.

### DISCUSSION THREAD

**Discussion by trocade** [Security-Onion-Solutions/securityonion#3729] [Unsupported Versions] [answered]
I noted when I tried to install a 2.3.40 sensor from scratch that it asks about proxy for installation, if I don't add a proxy it hangs for a long time and finally times out with (of course this node does not have internet connection)

![image](https://user-images.githubusercontent.com/9382220/113297938-8c1ba280-92fb-11eb-82fe-04fc06dfb2d6.png)

It seems that the install will continue after this.
Two questions:
1. Why is this check performed, and what is the problem if it fails? (based on my install, nothing)
2. The timeout could use some trimming :)

Also, since this is asked later on in the process:
![image](https://user-images.githubusercontent.com/9382220/113299285-026cd480-92fd-11eb-8277-95e33d0eae64.png)

**Comment by rwwiv** [comment, marked as answer]
#### To answer your questions:
1. This check being performed after answering no is a bug, and will be fixed in the next release. (See commit 88c565f)
2. Not really a question 😉 but it does seem like the default timeout of 2 minutes is a little too long and could probably use some trimming, yes. Expect to see this as well.

#### For some additional clarity:

The initial proxy question is asking whether you'd like to set a proxy for traffic that includes, but isn't limited to, OS package updates (see #529 and the corresponding PR #3503 for what exactly is being configured to run through the proxy). 

The later question is referring _only_ to OS package updates (i.e. `yum` or `apt`). Choosing to use the manager as the proxy for these updates overrides the general proxy chosen previously, if one was in fact set up. 

We agree that this is a little confusing, and are discussing what changes to make so it's more clear what's being asked in each case.

**Reply by trocade** [reply]
I see... :) My understanding is that it is optional and not something that will break the installation in any way.

Suggestion:
After configuring the manger ip/host etc then maybe ask for os updates via manger or direct, and if the answer is direct then ask for proxy? :)

**Reply by rwwiv** [reply]
That is correct, if the test fails you can just skip and move on. But it's definitely not intended for you to need to sit and wait for setup to fail that test if you chose not to set up a proxy.

Thanks for the suggestion! That's one of the options we're looking at, I'll make sure to give an update here once we make a decision.
